ci_tutorial/packages_pipeline/multi_configuration_lockfile.rst

```diff
@@ -0,0 +1,147 @@
+
+The way to avoid this discrepancy in dependencies is to force the usage of the same dependencies versions and revisions, something that can be done with :ref:`lockfiles<tutorial_versioning_lockfiles>`.
+
+Creating and applying lockfiles is relatively straightforward. The process of creating and promoting the configurations will be identical to the previous section, but just applying the lockfiles.
+
+Creating the lockfile
+---------------------
+
+Let's make sure as usual that we start from a clean state:
+
+.. code-block:: bash
+
+    $ conan remove "*" -c  # Make sure no packages from last run
+    $ conan remote remove "*"  # Make sure no other remotes defined
+    # Add develop repo, you might need to adjust this URL
+    $ conan remote add develop http://localhost:8081/artifactory/api/conan/develop
+
+
+Then we can create the lockfile ``conan.lock`` file:
+
+.. code-block:: bash
+
+    # Capture a lockfile for the Release configuration
+    $ conan lock create . -s build_type=Release --lockfile-out=conan.lock
+    # extend the lockfile so it also covers the Debug configuration
+    # in case there are Debug-specific dependencies
+    $ conan lock create . -s build_type=Debug --lockfile=conan.lock --lockfile-out=conan.lock
+
+Note that different configurations, using different profiles or settings could result in different dependency graphs. A lockfile file can be used to lock the different configurations, but it is important to iterate the different configurations/profiles and capture their information in the lockfile.
+
+.. note::
+
+    The ``conan.lock`` is the default argument, and if a ``conan.lock`` file exists, it might be automatically used by ``conan install/create`` and other graph commands. This can simplify many of the commands, but this tutorial is showing the full explicit commands for clarity and didactical reasons.
+
+The ``conan.lock`` file can be inspected, it will be something like:
+
+.. code-block:: json
+
+    {
+        "version": "0.5",
+        "requires": [
+            "mathlib/1.0#f2b05681ed843bf50d8b7b7bdb5163ea%1724319985.398"
+        ],
+        "build_requires": [],
+        "python_requires": [],
+        "config_requires": []
+    }
+    
+As we can see, it is locking the ``mathlib/1.0`` dependency version and revision.
+
+
+With the lockfile, the creating of the different configurations is exactly the same, but providing the ``--lockfile=conan.lock`` argument to the ``conan create`` step, it will guarantee that ``mathlib/1.0#f2b05681ed843bf50d8b7b7bdb5163ea`` will always be the exact dependency used, irrespective if there exist new ``mathlib/1.1`` versions or new revisions available. The following builds could be launched in parallel but executed at different times, and still they will always use the same ``mathlib/1.0`` dependency:
```

Comment:
```suggestion
With the lockfile, creating of the different configurations is exactly the same, but providing the ``--lockfile=conan.lock`` argument to the ``conan create`` step, it will guarantee that ``mathlib/1.0#f2b05681ed843bf50d8b7b7bdb5163ea`` will always be the exact dependency used, irrespective if there exist new ``mathlib/1.1`` versions or new revisions available. The following builds could be launched in parallel but executed at different times, and still they will always use the same ``mathlib/1.0`` dependency:
```